app/bundles/LeadBundle/Services/AnonymizeContactCompanyData.php

```diff
@@ -0,0 +1,378 @@
+use Mautic\LeadBundle\Model\FieldModel;
+use Psr\Log\LoggerInterface;
+
+class AnonymizeContactCompanyData
+{
+    public const COLUMNS_NOT_ACCEPTED = ['submission_id', 'form_id'];
+    /**
+     * @var array<string, string>
+     */
+    public const COMPANY_FIELDS_TO_COLUMNS =
+        [
+            'companyaddress1'    => 'address1',
+            'companyaddress2'    => 'address2',
+            'companycity'        => 'city',
+            'companystate'       => 'state',
+            'companyzip'         => 'zip',
+            'companycountry'     => 'country',
+            'companyphone'       => 'phone',
+            'companyfax'         => 'fax',
+            'companywebsite'     => 'website',
+            'companyemail'       => 'email',
+            'companyname'        => 'name',
+            'companydescription' => 'description',
+            'companyindustry'    => 'industry',
+            'companyemployees'   => 'employees',
+            'companyrevenue'     => 'revenue',
+            'companystatus'      => 'status',
+            'companytype'        => 'type',
+            'companyalias'       => 'alias',
+        ];
+
+    public function __construct(
+        private readonly FieldModel $fieldModel,
+        private readonly LoggerInterface $logger,
+        private readonly EmailModel $emailModel,
+        private readonly SubmissionModel $submissionModel,
+        private readonly EntityManagerInterface $entityManager,
+        private readonly AuditLogModel $auditLogModel,
+        private readonly CompanyModel $companyModel,
+    ) {
+    }
+
+    /**
+     * @param array<Lead>|array<Company> $leadsCompanies
+     *
+     * @return array<mixed>
+     */
+    public function setHashes(array $leadsCompanies, LeadField $field, bool $pseudonymize): array
+    {
+        foreach ($leadsCompanies as $key => $leadCompany) {
+            if (!method_exists($leadCompany, 'getField')) {
+                continue;
+            }
+            $leadField = false;
+            if ($leadCompany instanceof Company && 'company' === $field->getObject()) {
+                $leadField = $leadCompany->getField($field->getAlias());
+                if (false === $leadField) {
+                    $alias          = self::COMPANY_FIELDS_TO_COLUMNS[$field->getAlias()] ?? $field->getAlias();
+                    $leadFieldValue = $leadCompany->getFieldValue($alias);
+                    if (property_exists($leadCompany, $alias) && null !== $leadFieldValue) {
+                        $tempLeadField = $leadCompany->{'get'.ucfirst($alias)}();
+                        if (empty($tempLeadField)) {
+                            continue;
+                        }
+                        unset($leadField);
+                        $leadField['value'] = $tempLeadField;
+                        $leadField['type']  = $field->getType();
+                    } else {
+                        continue;
+                    }
+                }
+            }
```

Comment:
Do not see reason for this complexity of code. If we have company and alias we can use https://github.com/mautic/mautic/blob/7.x/app/bundles/LeadBundle/Model/CompanyModel.php#L235 for example